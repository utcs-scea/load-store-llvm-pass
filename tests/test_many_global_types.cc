#include <stdio.h>

int global_a = 5;
float global_b = 3.14;
char global_c = 'X';

void print_globals() {
  printf("global_a: %d, global_b: %.2f, global_c: %c\n", global_a, global_b, global_c);
}

int main() {
  print_globals();
  global_a *= 2;
  global_b += 1.0;
  global_c = 'Y';
  print_globals();
  return 0;
}
