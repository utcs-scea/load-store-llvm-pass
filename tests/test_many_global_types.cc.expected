   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_float32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_int8() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
global_a: 5, global_b: 3.14, global_c: X
   >> globalify() invoked
   >> __pando__replace_load_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_int32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_load_float32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_float32() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> __pando__replace_store_int8() invoked
   >> check_if_global() invoked
   >> deglobalify() invoked
   >> globalify() invoked
   >> deglobalify() invoked
global_a: 10, global_b: 4.14, global_c: Y
